File: android/versioned-abis/expoview-abi28_0_0/src/main/java/abi28_0_0/host/exp/exponent/modules/api/components/gesturehandler/OnTouchEventListener.java
package abi28_0_0.host.exp.exponent.modules.api.components.gesturehandler;

import android.view.MotionEvent;

import abi28_0_0.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler;

public interface OnTouchEventListener<T extends GestureHandler> {
  void onTouchEvent(T handler, MotionEvent event);
  void onStateChange(T handler, int newState, int oldState);
}

File: android/versioned-abis/expoview-abi44_0_0/src/main/java/abi44_0_0/host/exp/exponent/modules/api/components/gesturehandler/OnTouchEventListener.kt
package abi44_0_0.host.exp.exponent.modules.api.components.gesturehandler

import android.view.MotionEvent

interface OnTouchEventListener {
  fun <T : GestureHandler<T>> onHandlerUpdate(handler: T, event: MotionEvent)
  fun <T : GestureHandler<T>> onStateChange(handler: T, newState: Int, oldState: Int)
  fun <T : GestureHandler<T>> onTouchEvent(handler: T)
}
